src/components/statusHandler/StatusHandler.tsx

```diff
@@ -2,23 +2,45 @@ import React from "react";
 import { MadieAlert } from "@madie/madie-design-system/dist/react";
 import "./StatusHandler.scss";
 
-const StatusHandler = ({ error, errorMessage, testDataId }) => {
+const StatusHandler = ({ error, errorMessages, testDataId }) => {
   {
     if (error) {
-      if (errorMessage) {
-        return (
-          <MadieAlert
-            data-testid="generic-error-text-header"
-            type="error"
-            content={
-              <h3 aria-live="polite" role="alert" data-testid={testDataId}>
-                {errorMessage}
-              </h3>
-            }
-            canClose={false}
-          />
-        );
+      if (errorMessages) {
+        if (errorMessages.length === 1) {
+          return (
+            <MadieAlert
+              data-testid="generic-error-text-header"
+              type="error"
+              content={
+                <h3 aria-live="polite" role="alert" data-testid={testDataId}>
+                  {errorMessages}
+                </h3>
+              }
+              canClose={false}
+            />
+          );
+        } else if (errorMessages.length > 1) {
+          const mappedMessages = errorMessages.map(
+            (em: string, index: number) => <ol key={index}>{em}</ol>
+          );
+          return (
+            <MadieAlert
+              type="error"
+              content={
+                <div aria-live="polite" role="alert" data-testid={testDataId}>
+                  <h3>{errorMessages.length} errors were found</h3>
+                  <ul data-testid="generic-fail-text-list">{mappedMessages}</ul>
+                </div>
+              }
+              canClose={false}
+            />
+          );
+        } else {
+          return <div />;
+        }
       }
+    } else {
+      return <div />;
```

Comment:
Also is it possible to have a default return statement, as all our "else" block has the same content being returned?

Reply:
@RohitKandimalla Thanks a lot Rohit, just pushed in the change.